velox/functions/sparksql/ToJson.h

```diff
@@ -0,0 +1,398 @@
+void toJson<TypeKind::ROW>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options,
+    bool isMapKey) {
+  auto rowType = std::static_pointer_cast<const RowType>(input.type());
+  auto row = input.castTo<DynamicRow>();
+  result.append(isMapKey ? "[" : "{");
+  bool commaBefore = false;
+  for (int i = 0; i < rowType->size(); i++) {
+    auto data = row.at(i);
+    if (data.has_value()) {
+      if (commaBefore) {
+        result.append(",");
+      }
+      if (!isMapKey) {
+        result.append("\"");
+        result.append(rowType->nameOf(i));
+        result.append("\":");
+      }
+      VELOX_DYNAMIC_TYPE_DISPATCH_ALL(
+          toJson, data->kind(), data.value(), result, options, isMapKey);
+      commaBefore = true;
+    } else if (isMapKey) {
+      if (commaBefore) {
+        result.append(",");
+      }
+      result.append("null");
+      commaBefore = true;
+    } else if (!options.ignoreNullFields) {
+      if (commaBefore) {
+        result.append(",");
+      }
+      result.append("\"");
+      result.append(rowType->nameOf(i));
+      result.append("\":null");
+      commaBefore = true;
+    }
+  }
+  result.append(isMapKey ? "]" : "}");
+}
+
+template <>
+void toJson<TypeKind::ARRAY>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options,
+    bool isMapKey) {
+  auto arrayView = input.castTo<Array<Any>>();
+  result.append("[");
+  for (int i = 0; i < arrayView.size(); i++) {
+    if (i > 0) {
+      result.append(",");
+    }
+    if (arrayView[i].has_value()) {
+      auto element = arrayView[i].value();
+      VELOX_DYNAMIC_TYPE_DISPATCH_ALL(
+          toJson, element.kind(), element, result, options, isMapKey);
+    } else {
+      result.append("null");
+    }
+  }
+  result.append("]");
+}
+
+template <>
+void toJson<TypeKind::MAP>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options,
+    bool isMapKey) {
+  auto mapView = input.castTo<Map<Any, Any>>();
+  result.append("{");
+  for (int i = 0; i < mapView.size(); i++) {
+    if (i > 0) {
+      result.append(",");
+    }
+    auto element = mapView.atIndex(i);
+    auto key = element.first;
+    auto value = element.second;
+    result.append("\"");
+    VELOX_DYNAMIC_TYPE_DISPATCH_ALL(
+        toJson, key.kind(), key, result, options, true);
+    result.append("\":");
+    if (value.has_value()) {
+      VELOX_DYNAMIC_TYPE_DISPATCH_ALL(
+          toJson,
+          value.value().kind(),
+          value.value(),
+          result,
+          options,
+          isMapKey);
+    } else {
+      result.append("null");
+    }
+  }
+  result.append("}");
+}
+
+template <typename T>
+struct ToJsonFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& inputTypes,
+      const core::QueryConfig& config,
+      const void* /*input*/,
+      const void* /*timeZone*/ = nullptr) {
```

Comment:
A better approach would be to implement two initializing methods, and calls one from the other with `timeZone` being nullptr.